packages/mana/persitablebase.go

```diff
@@ -0,0 +1,183 @@
+package mana
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/iotaledger/hive.go/stringify"
+
+	"github.com/iotaledger/hive.go/identity"
+	"github.com/iotaledger/hive.go/marshalutil"
+	"github.com/iotaledger/hive.go/objectstorage"
+)
+
+type PersistableBaseMana struct {
+	objectstorage.StorableObjectFlags
+	ManaType           Type
+	BaseMana1          float64
+	EffectiveBaseMana1 float64
+	BaseMana2          float64
+	EffectiveBaseMana2 float64
+	LastUpdated        time.Time
+	NodeID             identity.ID
+
+	bytes []byte
+}
+
+// String returns a human readable version of the PersistableBaseMana.
+func (persistableBaseMana *PersistableBaseMana) String() string {
+	return stringify.Struct("PersistableBaseMana",
+		stringify.StructField("ManaType", fmt.Sprint(persistableBaseMana.ManaType)),
+		stringify.StructField("BaseMana1", fmt.Sprint(persistableBaseMana.BaseMana1)),
+		stringify.StructField("EffectiveBaseMana1", fmt.Sprint(persistableBaseMana.EffectiveBaseMana1)),
+		stringify.StructField("BaseMana2", fmt.Sprint(persistableBaseMana.BaseMana2)),
+		stringify.StructField("EffectiveBaseMana2", fmt.Sprint(persistableBaseMana.EffectiveBaseMana2)),
+		stringify.StructField("LastUpdated", fmt.Sprint(persistableBaseMana.LastUpdated)),
+		stringify.StructField("NodeID", persistableBaseMana.NodeID.String()),
+	)
+}
+
+var _ objectstorage.StorableObject = &PersistableBaseMana{}
```

Comment:
what does this line do?

Reply:
It checks, at compile time, that `PersistableBaseMana` satisfy the `objectstorage.StorableObject` interface